5.4harry_is_rational_but_not_terrible.py

```diff
@@ -0,0 +1,27 @@
+import os
+
+
+def harry_chapters():
+    """
+    find and rename the correct number and name of each chapter
+    :return:
+    """
+    directory = r'potter'
+
+    # Get a list of all files in the directory
+    files = os.listdir(directory)
+
+    # Loop through each file and rename it
+    for filename in files:
+        new_name = ''
+        with open(directory + '\\' + filename) as file:
+            for i in range(9):
```

Comment:
Regarding these lines:
1. Use documentation and comments to clarify what's happening
2. Replace explicit numbers with variables
3. rename file,x,y,i